### PR TITLE
Translate predefined pronouns when changing the app's language.

### PR DESCRIPTION
**Pull request by HorusGoul**
### Details
Check out the conversation in https://github.com/Expensify/App/issues/5113

### Fixed Issues
$ https://github.com/Expensify/App/issues/5113

### Tests
1. Launch the app and log in
2. Click settings > profile
3. Change preferred pronouns to "Call me by name" and save
4. Click the back arrow
5. In preferences change language to Spanish
6. Click profile (Perfil)
7. The selected option should be "Llámame por mi nombre"

### QA Steps
1. Launch the app and log in
2. Click settings > profile
3. Change preferred pronouns to "Call me by name" and save
4. Click the back arrow
5. In preferences change language to Spanish
6. Click profile (Perfil)
7. The selected option should be "Llámame por mi nombre"

### Tested On

- [x] Web
- [x] Mobile Web
- [ ] Desktop
- [x] iOS
- [ ] Android

#### Web
https://user-images.githubusercontent.com/6759612/136174329-d392566f-fd0e-4e3f-8a01-3e7b49b10482.mp4

#### Mobile Web
https://user-images.githubusercontent.com/6759612/136179186-101f7513-4e69-4da1-8326-acca08d8b182.mp4

### CODE REVIEW

File: src/pages/settings/Profile/ProfilePage.js
Comment (Beamanator) [inline src/pages/settings/Profile/ProfilePage.js:82]: Since `selfSelectedPronouns` is now a boolean (not an array) I think we should change the variable name to indicate it's a boolean value. Maybe `hasSelfSelectedPronouns`?

File: src/pages/settings/Profile/ProfilePage.js
Comment (Beamanator) [inline src/pages/settings/Profile/ProfilePage.js:222]: Similar here, I think it makes sense to make the variable name more clearly a boolean, like `areSelfSelectedPronouns`

**Review by Beamanator — COMMENTED** [review]
Tested well! Just a little variable name change request, then we're good to go (after the n6 hold)

File: src/pages/settings/Profile/ProfilePage.js
Comment (HorusGoul) [reply src/pages/settings/Profile/ProfilePage.js]: Done!

File: src/pages/settings/Profile/ProfilePage.js
Comment (HorusGoul) [reply src/pages/settings/Profile/ProfilePage.js]: Done! I used `hasSelfSelectedPronouns` in both places.

**Review by Beamanator — DISMISSED** [review]
LGTM (after n-6 hold right?)

**Comment by HorusGoul** [comment]
> LGTM (after n-6 hold right?)

Yes!

**Comment by Beamanator** [comment]
Ooh! Would you mind merging `main` in here just to be super safe, then we test one last time before merge? :D

**Review by Beamanator — APPROVED** [review]
LGTM! Note: Here's my iOS test (you can put this in your OP if you want)

https://user-images.githubusercontent.com/3885503/139021388-367b58ef-1aa1-4588-83ea-b79f1d9a86e7.mov

**Pull request merged**
